packages/medusa/src/services/swap.ts

```diff
@@ -650,15 +702,17 @@ class SwapService extends BaseService {
 
       swap.cart_id = cart.id
 
-      const swapRepo = manager.getCustomRepository(this.swapRepository_)
       return await swapRepo.save(swap)
     })
   }
 
   /**
-   *@param {string} swapId - The id of the swap
+   * Register a cart completion
+   *
+   * @param swapId - The id of the swap
+   * @return swap related to the cart
    */
-  async registerCartCompletion(swapId) {
+  async registerCartCompletion(swapId: string): Promise<Swap | never> {
     return await this.atomicPhase_(async (manager) => {
       const swap = await this.retrieve(swapId, {
```

Comment:
suggestion: line 751 I would suggest that we call only one time the withTransaction just before the loop to avoid to instanciate the service on each loop.
```ts
const inventoryServiceTx = this.inventoryService_.withTransaction(manager)
```
And then we can use inventoryServiceTx. I suppose that everywhere in the code base we should be aware of that and do this kind of approach to avoid multiple instanciation each time. wdyt?